### PR TITLE
PHOENIX-6478 Add aliyunmaven repo for pentaho-aggdesigner-algorithm

### PR DESCRIPTION
**Pull request by richardantal**
Change-Id: I6beb2d95a63f8e3aebb22250918176329173066e

### CODE REVIEW

File: phoenix-hive-base/pom.xml [hunk 1]
```diff
@@ -583,4 +583,15 @@
       </plugins>
     </pluginManagement>
   </build>
+
+  <repositories>
+    <repository>
+      <id>saliyunmaven</id>
+      <name>aliyunmaven</name>
+      <url>https://maven.aliyun.com/repository/spring-plugin</url>
```
Comment (martin-g) [inline phoenix-hive-base/pom.xml:591]: Why do you prefer this repository than https://public.nexus.pentaho.org/repository/proxy-public-3rd-party-release ?

Comment (richardantal) [reply phoenix-hive-base/pom.xml:591]: The precommit test on your change takes way too much time. https://github.com/apache/phoenix-connectors/pull/53/checks 
I wanted to test if using this repo is faster. Locally it was better.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 30s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| -1 :x: |  mvninstall  |  23m 32s |  root in master failed.  |
| -1 :x: |  compile  |   0m 28s |  phoenix-hive-base in master failed.  |
| -1 :x: |  javadoc  |   0m 12s |  phoenix-hive-base in master failed.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  17m 31s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  6s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  6s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  2s |  The patch has no ill-formed XML file.  |
| -1 :x: |  javadoc  |   0m 24s |  phoenix-hive-base generated 6 new + 0 unchanged - 0 fixed = 6 total (was 0)  |
||| _ Other Tests _ |
| -1 :x: |  unit  |  55m 47s |  phoenix-hive-base in the patch failed.  |
| -1 :x: |  asflicense  |   0m 10s |  The patch generated 2 ASF License warnings.  |
|  |   | 100m  3s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-55/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix-connectors/pull/55 |
| Optional Tests | dupname asflicense javac javadoc unit xml compile |
| uname | Linux d21e0c4cb459 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-connectors-personality.sh |
| git revision | master / a883279 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| mvninstall | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-55/1/artifact/yetus-general-check/output/branch-mvninstall-root.txt |
| compile | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-55/1/artifact/yetus-general-check/output/branch-compile-phoenix-hive-base.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-55/1/artifact/yetus-general-check/output/branch-javadoc-phoenix-hive-base.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-55/1/artifact/yetus-general-check/output/diff-javadoc-javadoc-phoenix-hive-base.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-55/1/artifact/yetus-general-check/output/patch-unit-phoenix-hive-base.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-55/1/testReport/ |
| asflicense | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-55/1/artifact/yetus-general-check/output/patch-asflicense-problems.txt |
| Max. process+thread count | 1808 (vs. ulimit of 30000) |
| modules | C: phoenix-hive-base U: phoenix-hive-base |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-55/1/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   5m 28s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |  10m 52s |  Maven dependency ordering for branch  |
| -1 :x: |  mvninstall  |  16m 22s |  root in master failed.  |
| -1 :x: |  compile  |   0m 10s |  phoenix4-hive in master failed.  |
| -1 :x: |  compile  |   0m 11s |  phoenix4-hive-shaded in master failed.  |
| -1 :x: |  javadoc  |   0m 10s |  phoenix4-hive in master failed.  |
| -1 :x: |  javadoc  |   0m  9s |  phoenix4-hive-shaded in master failed.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |  20m 53s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 50s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 50s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  3s |  The patch has no ill-formed XML file.  |
| -1 :x: |  javadoc  |   0m 14s |  phoenix-hive-base_phoenix4-hive generated 3 new + 0 unchanged - 0 fixed = 3 total (was 0)  |
||| _ Other Tests _ |
| -1 :x: |  unit  |  40m 43s |  phoenix4-hive in the patch failed.  |
| +1 :green_heart: |  unit  |   5m 18s |  phoenix4-hive-shaded in the patch passed.  |
| -1 :x: |  asflicense  |   0m 17s |  The patch generated 1 ASF License warnings.  |
|  |   | 102m 36s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | phoenix.hive.HiveTezIT |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-55/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix-connectors/pull/55 |
| Optional Tests | dupname asflicense javac javadoc unit xml compile |
| uname | Linux 08e46d8fa2cd 4.15.0-142-generic #146-Ubuntu SMP Tue Apr 13 01:11:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-connectors-personality.sh |
| git revision | master / a883279 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| mvninstall | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-55/2/artifact/yetus-general-check/output/branch-mvninstall-root.txt |
| compile | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-55/2/artifact/yetus-general-check/output/branch-compile-phoenix-hive-base_phoenix4-hive.txt |
| compile | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-55/2/artifact/yetus-general-check/output/branch-compile-phoenix-hive-base_phoenix4-hive-shaded.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-55/2/artifact/yetus-general-check/output/branch-javadoc-phoenix-hive-base_phoenix4-hive.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-55/2/artifact/yetus-general-check/output/branch-javadoc-phoenix-hive-base_phoenix4-hive-shaded.txt |
| javadoc | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-55/2/artifact/yetus-general-check/output/diff-javadoc-javadoc-phoenix-hive-base_phoenix4-hive.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-55/2/artifact/yetus-general-check/output/patch-unit-phoenix-hive-base_phoenix4-hive.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-55/2/testReport/ |
| asflicense | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-55/2/artifact/yetus-general-check/output/patch-asflicense-problems.txt |
| Max. process+thread count | 1220 (vs. ulimit of 30000) |
| modules | C: phoenix-hive-base/phoenix4-hive phoenix-hive-base/phoenix4-hive-shaded U: phoenix-hive-base |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-55/2/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by stoty** [comment]
Can we check if the tests actually need this ?
Since we're not shipping the Hive parts, maybe we could just exclude it if doesn't break the tests.

**Comment by martin-g** [comment]
With the following change:

```
diff --git phoenix-hive-base/phoenix4-hive/pom.xml phoenix-hive-base/phoenix4-hive/pom.xml
index d08cf42..e3fb8a2 100644
--- phoenix-hive-base/phoenix4-hive/pom.xml
+++ phoenix-hive-base/phoenix4-hive/pom.xml
@@ -107,6 +107,25 @@
     </dependency>
   </dependencies>
 
+
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+      
+      <groupId>org.apache.calcite</groupId>
+      <artifactId>calcite-core</artifactId>
+      <version>1.10.0</version>
+      <exclusions>
+        <exclusion>
+
+      <groupId>org.pentaho</groupId>
+      <artifactId>pentaho-aggdesigner-algorithm</artifactId>
+        </exclusion>
+      </exclusions>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+
   <build>
     <plugins>
       <plugin>
```

the tests pass:

```

[INFO] --- maven-surefire-plugin:2.22.0:test (default-test) @ phoenix4-hive ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.phoenix.hive.PrimaryKeyDataTest
[INFO] Running org.apache.phoenix.hive.query.PhoenixQueryBuilderTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.092 s - in org.apache.phoenix.hive.PrimaryKeyDataTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.623 s - in org.apache.phoenix.hive.query.PhoenixQueryBuilderTest
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0
```

**Comment by richardantal** [comment]
IT tests were also successful.
Maybe we should just exclude it.

**Comment by stoty** [comment]
Sounds good to me.

**Comment by martin-g** [comment]
I've updated https://github.com/apache/phoenix-connectors/pull/53

**Pull request closed without merging**